### ISSUE TITLE
exceptions' stacktraces do not include the first line

### ISSUE THREAD

**Issue by ags799** [bbilger/jrestless#39]
My function is throwing an exception when its Jackson decoder tries to deserialize input JSON:

```
2017-05-15 22:47:56 <874465cf-39c0-11e7-8842-ab61849afd56> ERROR com.jrestless.core.container.JRestlessHandlerContainer:356 - container failure
Invalid date    at com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException.from(UnrecognizedPropertyException.java:62)
        at com.fasterxml.jackson.databind.DeserializationContext.handleUnknownProperty(DeserializationContext.java:834)
        at com.fasterxml.jackson.databind.deser.std.StdDeserializer.handleUnknownProperty(StdDeserializer.java:1093)
```

This specific exception is not the source of this issue. The issue is that the first line of the stacktrace is not being shown.

The relevant line seems to be [here](https://github.com/bbilger/jrestless/blob/master/core/jrestless-core-container/src/main/java/com/jrestless/core/container/JRestlessHandlerContainer.java#L354). Note that the first line of the stacktrace is written if we

```java
LOG.error("first line: {}", error.getMessage());
```

**Comment by bbilger** [comment]
Hi,
Not sure if I understand the issue here.
From the stack it looks like the JSON you try to deserialize contains an invalid/unexected date format.
What about registering a custom exception mapper like shown in #33? Would this help in your case?

**Comment by ags799** [comment]
The `Invalid date` bit is a bug in serverless: https://github.com/serverless/serverless/issues/3594

It should appear next to the first, missing line in the stacktrace.

Will try the approach in #33 for now, thanks.

**Comment by bbilger** [comment]
Hi @ags799,
I think that's not an issue with JRestless itself since if I check the logs in CloudWatch, I do see the exception's message just fine.
To me it looks like the `Invalid date` bug in the serverless framework swallows the exceptions message (i.e. first line, too).
Can you confirm this?

**Comment by bbilger** [comment]
according to serverless/serverless#3594 this should get fixed with serverless 1.15.0
will check once it's released
(but this issue is unrelated to JRestless)

**Comment by ags799** [comment]
agreed feel free to close
